diff -uNr a/RenderSystems/GL3Plus/include/windowing/win32/OgreWin32Window.h b/RenderSystems/GL3Plus/include/windowing/win32/OgreWin32Window.h
--- a/RenderSystems/GL3Plus/include/windowing/win32/OgreWin32Window.h	Fri Aug 28 21:31:58 2015
+++ b/RenderSystems/GL3Plus/include/windowing/win32/OgreWin32Window.h	Tue Sep 15 12:53:01 2015
@@ -92,7 +92,7 @@
         bool    mIsExternal;
         char*   mDeviceName;
         bool    mIsExternalGLControl;
-        bool    mIsExternalGLContext;
+        bool    mOwnsGLContext;
         bool    mSizing;
         bool    mClosed;
         bool    mHidden;
diff -uNr a/RenderSystems/GL3Plus/src/windowing/win32/OgreWin32Window.cpp b/RenderSystems/GL3Plus/src/windowing/win32/OgreWin32Window.cpp
--- a/RenderSystems/GL3Plus/src/windowing/win32/OgreWin32Window.cpp	Tue Sep 15 12:39:07 2015
+++ b/RenderSystems/GL3Plus/src/windowing/win32/OgreWin32Window.cpp	Tue Sep 15 12:35:22 2015
@@ -56,7 +56,7 @@
         mGlrc = 0;
         mIsExternal = false;
         mIsExternalGLControl = false;
-        mIsExternalGLContext = false;
+        mOwnsGLContext = true;
         mSizing = false;
         mClosed = false;
         mHidden = false;
@@ -164,6 +164,8 @@
 			}
 #endif
 
+
+
             if ((opt = miscParams->find("externalWindowHandle")) != end)
             {
                 mHWnd = (HWND)StringConverter::parseSizeT(opt->second);
@@ -177,11 +179,37 @@
                   mIsExternalGLControl = StringConverter::parseBool(opt->second);
                 }
             }
+
+            if ((opt = miscParams->find("currentGLContext")) != end )
+            {
+                if( StringConverter::parseBool(opt->second) )
+                {
+                    mGlrc = wglGetCurrentContext();
+
+                    if ( mGlrc )
+                    {
+                        mOwnsGLContext = false;
+                    }
+                    else
+                    {
+                        OGRE_EXCEPT(Exception::ERR_RENDERINGAPI_ERROR,
+                        "currentGLContext was specified with no current GL context", "Win32Window::create");
+                    }
+                }
+            }
             if ((opt = miscParams->find("externalGLContext")) != end)
             {
                 mGlrc = (HGLRC)StringConverter::parseUnsignedLong(opt->second);
+
                 if( mGlrc )
-                    mIsExternalGLContext = true;
+                {
+                    mOwnsGLContext = false;
+                }
+                else
+                {
+                    OGRE_EXCEPT(Exception::ERR_RENDERINGAPI_ERROR,
+                    "parsing the value of 'externalGLContext' failed: " + translateWGLError(), "Win32Window::create");
+                }
             }
 
             // window border style
@@ -449,7 +477,7 @@
             mHwGamma = testHwGamma;
             mFSAA = testFsaa;
         }
-        if (!mIsExternalGLContext)
+        if (mOwnsGLContext)
         {
             mGlrc = wglCreateContext(mHDC);
             if (!mGlrc)
@@ -639,7 +667,7 @@
         delete mContext;
         mContext = 0;
 
-        if (!mIsExternalGLContext && mGlrc)
+        if (mOwnsGLContext)
         {
             wglDeleteContext(mGlrc);
             mGlrc = 0;
