--- a/cmake/EigenConfigureTesting.cmake	2015-10-01 09:50:45.000000000 +0200
+++ b/cmake/EigenConfigureTesting.cmake	2015-10-01 09:57:10.000000000 +0200
@@ -39,7 +39,7 @@
 
 # copy ctest properties, which currently
 # o raise the warning levels
-configure_file(${CMAKE_BINARY_DIR}/DartConfiguration.tcl ${CMAKE_BINARY_DIR}/DartConfiguration.tcl)
+#configure_file(${CMAKE_BINARY_DIR}/DartConfiguration.tcl ${CMAKE_BINARY_DIR}/DartConfiguration.tcl)
 
 # restore default CMAKE_MAKE_PROGRAM
 set(CMAKE_MAKE_PROGRAM ${CMAKE_MAKE_PROGRAM_SAVE})
