# HG changeset patch
# Parent  9e3c199317b69bb817810c53aaf7f051498933a6
[mq]: AddedVC2013.diff

diff --git a/Makefile.mak b/Makefile.mak
--- a/Makefile.mak
+++ b/Makefile.mak
@@ -26,7 +26,7 @@
 
 VER = 0.9.20
 ARCH = win64
-DEBUG = no
+DEBUG = Debug
 BRANCH = v$(VER)-$(ARCH)
 MSVSVER =
 MSVCROOT = $(VCINSTALLDIR)
@@ -147,7 +147,7 @@
          /INCREMENTAL:NO \
          /MANIFEST
 
-!if "$(DEBUG)" != "no"
+!if "$(DEBUG)" != "Release"
 !message === DEBUGGING ON
 CFLAGS = $(CFLAGS) /Z7 /D"DEBUG"
 LFLAGS = $(LFLAGS) /DEBUG
@@ -356,8 +356,10 @@
      $(cmdexe)
 !endif
 
+!if "$(MEX)" == ""
 BUILD_MEX=@echo .... CC [MEX] $(@) && \
 	$(MEX) $(MEX_FLAGS) "$(<)" -output $(@)
+!endif
 
 # --------------------------------------------------------------------
 #                                                    Maintenance rules
@@ -412,13 +414,13 @@
 
 # create directory if missing
 $(bindir) :
-	mkdir $(bindir)
+	md $(bindir)
 
 $(objdir) :
-	mkdir $(objdir)
+	md $(objdir)
 
 $(mexdir) :
-	mkdir $(mexdir)
+	md $(mexdir)
 
 # --------------------------------------------------------------------
 #                                      Rules to compile the VLFeat DLL
