--- ./modules/legacy/src/calibfilter.cpp.old	2015-02-27 14:33:00.983420840 +0100
+++ ./modules/legacy/src/calibfilter.cpp	2015-02-27 14:36:54.449786318 +0100
@@ -95,7 +95,7 @@
 
     Stop();
 
-    if (latestPoints != NULL)
+    //if (latestPoints != NULL)
     {
         for( i = 0; i < MAX_CAMERAS; i++ )
             cvFree( latestPoints + i );
@@ -523,7 +523,7 @@
         return;
     }
 
-    if( latestCounts )
+    //if( latestCounts )
     {
         for( i = 0; i < cameraCount; i++ )
         {
