diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9646244..1410a8f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -152,11 +152,9 @@ endif()
 
 if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
   SET(CMAKE_COMPILER_IS_CLANG 1)
-  if("${CMAKE_C_FLAGS}" STREQUAL "")
-    SET(CMAKE_C_FLAGS "-Qunused-arguments")
-  endif()
+
   if("${CMAKE_CXX_FLAGS}" STREQUAL "")
-    SET(CMAKE_CXX_FLAGS "-ftemplate-depth=1024 -Qunused-arguments -Wno-invalid-offsetof ${SSE_FLAGS}") # Unfortunately older Clang versions do not have this: -Wno-unnamed-type-template-args
+    SET(CMAKE_CXX_FLAGS "-ftemplate-depth=1024 -Wno-invalid-offsetof ${SSE_FLAGS}") # Unfortunately older Clang versions do not have this: -Wno-unnamed-type-template-args
     if(APPLE AND WITH_CUDA AND CUDA_FOUND)
       SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libstdc++")
     endif()
