--- ./CMakeLists.txt	2015-06-24 16:33:43.000000000 +0200
+++ ./CMakeLists.txt	2015-07-07 11:42:03.918340828 +0200
@@ -64,6 +64,10 @@
     SET(CMAKE_LIBRARY_PATH ${CMAKE_LIBRARY_PATH} ${GLUT_PATH}/lib)
 ENDIF()
 
+if(ANDROID)
+    include_directories("${CMAKE_INSTALL_PREFIX}/include/")
+endif()
+
 ##LOOK FOR OPENGL AND GLUT
 #FIND OPENGL LIBRARY. In Android, there is no need since it is included
 IF(NOT ANDROID_CREATION)
