diff -uNr b/CMakeLists.txt pangolin/CMakeLists.txt
--- b/CMakeLists.txt	2016-06-17 11:04:36.239522738 +0200
+++ pangolin/CMakeLists.txt	2016-06-17 11:05:34.549595443 +0200
@@ -22,9 +22,10 @@
 endif()
 
 if(CPP11_NO_BOOST AND NOT MSVC)
-    set( CMAKE_CXX_FLAGS "-std=c++0x -Wall ${CMAKE_CXX_FLAGS}" )
     if(_CLANG_)
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall")
+    else()
+        set(CMAKE_CXX_FLAGS "-std=c++0x -Wall ${CMAKE_CXX_FLAGS}")
     endif()
 endif()
 
