diff --git a/src/markerdetector.h b/src/markerdetector.h
index c9bccba..13cacdd 100644
--- a/src/markerdetector.h
+++ b/src/markerdetector.h
@@ -294,6 +294,15 @@ class ARUCO_EXPORTS MarkerDetector {
     static void glGetProjectionMatrix(CameraParameters &CamMatrix, cv::Size orgImgSize, cv::Size size, double proj_matrix[16], double gnear, double gfar,
                                       bool invert = false) throw(cv::Exception);

+    /**
+    * Set the percentage of border around image limits in which corners are not allowed to be detected.
+    * @param value : percentage (default 0.025)
+    */
+    inline void setBorderDistance(float value)
+    {
+        _borderDistThres = value;
+    }
+
   private:
     bool warp_cylinder(cv::Mat &in, cv::Mat &out, cv::Size size, MarkerCandidate &mc) throw(cv::Exception);
     /**
